### PR TITLE
Upgrade Wildfly Swarm to Thorntail

### PR DESCRIPTION
**Pull request by juangon**
Hi!

I upgraded these tests so they can pass with new Thorntail versions.

Want to highlight 2 changes made and their reasons:

* Added better ExecutionException check. As stated in https://github.com/eclipse/microprofile-fault-tolerance/blob/master/spec/src/main/asciidoc/asynchronous.asciidoc#exception-handling, ExecutionException can wrap the exceptions from Bulkhead, Timeout, etc.
* In circuitbreaker tests, I added 2 changes:
  * Added one more iteration to open circuit. This was clarified as part of this: https://github.com/eclipse/microprofile-fault-tolerance/issues/341 (opened by @mkouba)
  * As there could be problems when using concurrent request (it may happen that circuit isn't open when throwing the exact number of request to get expected number of requests) I changed requests to be sequential by using newSingleThread executor.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 		<payara.version>5.182-SNAPSHOT</payara.version>
 		<payara.micro.version>5.182-SNAPSHOT</payara.micro.version>
 		<liberty.version>17.0.0.4</liberty.version>
-		<wildfly.version>2018.4.1</wildfly.version>
+		<thorntail.version>2.3.0.Final-SNAPSHOT</thorntail.version>
```
Comment (Ladicek) [inline pom.xml:38]: Shouldn't be merged now then.

**Comment by Ladicek** [comment]
> As there could be problems when using concurrent request (it may happen that circuit isn't open when throwing the exact number of request to get expected number of requests) I changed requests to be sequential by using newSingleThread executor.

I think you should enable `thorntail.microprofile.fault-tolerance.synchronous-circuit-breaker` -- that enables 100% compliant behavior at the expense of probably worse performance.

File: pom.xml (hunk 1 above)
Comment (juangon) [reply pom.xml:38]: Ok better wait to 2.3.0.Final right?

**Comment by juangon** [comment]
> > As there could be problems when using concurrent request (it may happen that circuit isn't open when throwing the exact number of request to get expected number of requests) I changed requests to be sequential by using newSingleThread executor.
> 
> I think you should enable `thorntail.microprofile.fault-tolerance.synchronous-circuit-breaker` -- that enables 100% compliant behavior at the expense of probably worse performance.

It's enabled by default. I enabled it just in case and doesn't work.

**Comment by juangon** [comment]
I am gonna send PR whenever 2.3.0.Final is out so we don't use SNAPSHOT versions.

Thanks @Ladicek for the review!

File: pom.xml (hunk 1 above)
Comment (Ladicek) [reply pom.xml:38]: Yea, if it doesn't work with 2.2.1.Final, then we need to wait for 2.3.0.Final.

**Comment by arjantijms** [comment]
Thanks @Ladicek indeed!

@juangon Will be looking forward to your next PR, thx!

**Comment by juangon** [comment]
Resent here: https://github.com/javaee-samples/microprofile1.2-samples/pull/7

**Pull request closed without merging**